snuba/datasets/metrics_processor.py

```diff
@@ -78,3 +78,16 @@ def _process_values(self, message: Mapping[str, Any]) -> Mapping[str, Any]:
             value, (int, float)
         ), "Illegal value for counter value. Int/Float expected {value}"
         return {"value": value}
+
+
+class DistributionsMetricsProcessor(MetricsProcessor):
+    def _should_process(self, message: Mapping[str, Any]) -> bool:
+        return message["type"] is not None and message["type"] == "d"
+
+    def _process_values(self, message: Mapping[str, Any]) -> Mapping[str, Any]:
+        values = message["value"]
```

Comment:
Is there some limit on the size of this array?

Reply:
Not at this stage. Basically the current limit is the size of the Kafka message.
Later this quarter we will have to work with the ingest team to optimize the size of the batch and the encoding to keep them small (reducing the max size does not make the bucket smaller as there would be more buckets).
I think the main parameters we will have to deal with will be the size of the bucket at relay (which drives the message size) and the batch size in the consumer, which drives the size and the number of data parts in clickhouse. 
But this code would most likely break arrays that are too long int multiple INSERT statements instead of rejecting them. 
Though this has the drawback of increasing the parts in clickhouse (which increases the files and adds more merge work). 
So we will have a better answer in july I guess

Reply:
OK. It sounds like at least there is some thinking around this.